sql/moz-fx-data-shared-prod/accounts_backend_derived/monitoring_db_counts_v1/query.sql

```diff
@@ -0,0 +1,300 @@
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'recovery_codes' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_recovery_codes_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'security_events' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_security_events_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'sent_emails' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_sent_emails_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'session_tokens' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_session_tokens_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'signin_codes' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_signin_codes_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'totp' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_totp_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'unblock_codes' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_unblock_codes_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'unverified_tokens' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_unverified_tokens_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+  SELECT
+    'verification_reminders' AS table_name,
+    COUNT(*) AS total_rows
+  FROM
+    `moz-fx-data-shared-prod.accounts_db_external.fxa_verification_reminders_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+      @as_of_date,
+      'UTC'
+    )
+  UNION ALL
+    (
+      SELECT
+        "accounts_with_secondary_emails" AS table_name,
+        COUNT(
+          DISTINCT `moz-fx-data-shared-prod.accounts_db_external.fxa_accounts_v1`.uid
+        ) AS total_rows
+      FROM
+        `moz-fx-data-shared-prod.accounts_db_external.fxa_accounts_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+          @as_of_date,
+          'UTC'
+        )
+      JOIN
+        `moz-fx-data-shared-prod.accounts_db_external.fxa_emails_v1` FOR SYSTEM_TIME AS OF TIMESTAMP(
+          @as_of_date,
+          'UTC'
+        )
+        ON `moz-fx-data-shared-prod.accounts_db_external.fxa_accounts_v1`.uid = `moz-fx-data-shared-prod.accounts_db_external.fxa_emails_v1`.uid
+      WHERE
+        `moz-fx-data-shared-prod.accounts_db_external.fxa_emails_v1`.isPrimary = FALSE
```

Comment:
This would unfortunately fail on formatting. Sqlgot seems not to be able to handle table aliases combined with another expression following them (`FOR SYSTEM_TIME` here).

Reply:
Dang. I'd suggest [reporting](https://github.com/tobymao/sqlglot/issues) that SQLGlot issue, as they're very responsive and have fixed every issue I've reported to them (often the same day on `main`, to be included in the next release).

Reply:
Another option is if you don't quote the whole table identifier (e.g. `` `moz-fx-data-shared-prod`.accounts_db_external.fxa_accounts_v1``) then BigQuery allows you to refer to just the final table name segment, like an automatic alias. (though I do generally prefer to quote the entire table identifier)